File: GameServer/Content/Data/LeagueSandbox-Default/Champions/Lucian/W.lua
Vector2 = require 'Vector2' -- include 2d vector lib 

function onFinishCasting()
	local current = Vector2:new(owner.X, owner.Y)
	local to = (Vector2:new(spell.X, spell.Y) - current):normalize()
	local range = to * 900
	local trueCoords = current + range
	printChat("You used W");
	addProjectile("LucianWMissile", trueCoords.x, trueCoords.y)
end
function applyEffects()
	local damage = 20+spellLevel*40+owner:GetStats().AbilityPower.Total*0.9
    dealMagicalDamage(damage)
end

File: Content/LeagueSandbox-Scripts/Champions/Lucian/W.cs
using System.Numerics;
using GameServerCore.Domain.GameObjects;
using LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI;
using LeagueSandbox.GameServer.GameObjects.Missiles;
using GameServerCore.Domain;
using LeagueSandbox.GameServer.Scripting.CSharp;

namespace Spells
{
    public class LucianW : IGameScript
    {
        public void OnActivate(IObjAiBase owner)
        {
        }

        public void OnDeactivate(IObjAiBase owner)
        {
        }

        public void OnStartCasting(IObjAiBase owner, ISpell spell, IAttackableUnit target)
        {
        }

        public void OnFinishCasting(IObjAiBase owner, ISpell spell, IAttackableUnit target)
        {
            var current = new Vector2(owner.X, owner.Y);
            var to = Vector2.Normalize(new Vector2(spell.X, spell.Y) - current);
            var range = to * 900;
            var trueCoords = current + range;

            spell.AddProjectile("LucianWMissile", owner.X, owner.Y, trueCoords.X, trueCoords.Y);
        }

        public void ApplyEffects(IObjAiBase owner, IAttackableUnit target, ISpell spell, IProjectile projectile)
        {
            //float damage = 20 + (spellLevel * 40) + owner.GetStats().AbilityPower.Total * 0.9;
            //dealMagicalDamage(damage);
        }

        public void OnUpdate(double diff)
        {
        }
    }
}
